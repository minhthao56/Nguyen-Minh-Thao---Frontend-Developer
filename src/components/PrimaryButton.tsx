import { PropsWithChildren } from "react";

export interface PrimaryButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
    PropsWithChildren {
  color?: "black" | "white";
  className?: string;
}

export default function PrimaryButton({
  className,
  color = "black",
  ...rest
}: PrimaryButtonProps) {
  return (
    <button
      type="submit"
      className={`text-${
        color === "black" ? "white" : "black"
      } bg-${color} focus:ring-4 focus:outline-none focus:ring-${color} font-medium rounded-lg text-sm w-full px-5 py-2.5 text-center ${className}`}
      {...rest}
    >
      {rest.children}
    </button>
  );
}
