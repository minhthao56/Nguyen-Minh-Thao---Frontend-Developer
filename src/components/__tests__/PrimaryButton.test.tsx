import { render, screen } from "@testing-library/react";
import PrimaryButton, { PrimaryButtonProps } from "../PrimaryButton";

describe("<PrimaryButton/>", () => {
  const renderComponent = (props?: Partial<PrimaryButtonProps>) => {
    return render(<PrimaryButton {...props}>{props?.children}</PrimaryButton>);
  };

  it("should match to snapshot with black", () => {
    const { container } = renderComponent({ children: "Name", color: "black" });
    expect(container).toMatchSnapshot();
    expect(screen.getByText("Name")).toBeInTheDocument();
  });

  it("should match to snapshot with white", () => {
    const { container } = renderComponent({ children: "Name", color: "white" });
    expect(container).toMatchSnapshot();
    expect(screen.getByText("Name")).toBeInTheDocument();
  });
});
