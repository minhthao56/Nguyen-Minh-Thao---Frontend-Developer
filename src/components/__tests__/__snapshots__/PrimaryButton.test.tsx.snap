// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryButton/> should match to snapshot with black 1`] = `
<div>
  <button
    class="text-white bg-black focus:ring-4 focus:outline-none focus:ring-black font-medium rounded-lg text-sm w-full px-5 py-2.5 text-center undefined"
    type="submit"
  >
    Name
  </button>
</div>
`;

exports[`<PrimaryButton/> should match to snapshot with white 1`] = `
<div>
  <button
    class="text-black bg-white focus:ring-4 focus:outline-none focus:ring-white font-medium rounded-lg text-sm w-full px-5 py-2.5 text-center undefined"
    type="submit"
  >
    Name
  </button>
</div>
`;
